lib/tqueue.ml

```diff
@@ -0,0 +1,65 @@
+type 'a t = {
```

Comment:
Why `Tqueue` instead of `Queue`? 🤔

Reply:
Queue's are not thread safe, I implemented a thread-safe queue library that uses mutexes to protect read/write operations, and a condition var to allow blocking until the queue has an element.